gossipd/gossip.c:1702:7: error: implicit declaration of function ‘structeq’

Duplicate of comment https://github.com/ElementsProject/lightning/pull/1618#issuecomment-403248725 in closed entry.

### Issue and Steps to Reproduce
<!-- Describe your issue and tell us how to reproduce it (include any useful information). -->
$ make
...
cc -DBINTOPKGLIBEXECDIR='"'../libexec/c-lightning'"' -Werror -Wall -Wundef -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes -Wold-style-definition -std=gnu11 -g -fstack-protector -I ccan -I external/libsodium/src/libsodium/include -I external/libwally-core/include/ -I external/libwally-core/src/secp256k1/include/ -I external/jsmn/ -I external/libbase58/ -I external/libbacktrace/ -I external/libbacktrace-build -I . -I/usr/local/include    -DSHACHAIN_BITS=48 -DJSMN_PARENT_LINKS  -DCOMPAT_V052=1 -DBINTOPKGLIBEXECDIR='"'../libexec/c-lightning'"'  -c -o gossipd/gossip.o gossipd/gossip.c
gossipd/gossip.c: In function ‘find_local_update’:
gossipd/gossip.c:1702:7: error: implicit declaration of function ‘structeq’ [-Werror=implicit-function-declaration]
   if (structeq(scid, &i->scid))
       ^
cc1: all warnings being treated as errors
<builtin>: recipe for target 'gossipd/gossip.o' failed
make: *** [gossipd/gossip.o] Error 1
rm external/libwally-core/src/secp256k1/libsecp256k1.la
$
### `getinfo` output

--- a/gossipd/gossip.c
+++ b/gossipd/gossip.c
@@ -1699,7 +1699,7 @@ static struct local_update *find_local_update(struct daemon *daemon,
 	struct local_update *i;
 
 	list_for_each(&daemon->local_updates, i, list) {
-		if (structeq(scid, &i->scid))
+		if (short_channel_id_eq(scid, &i->scid))
 			return i;
 	}
 	return NULL;

